js/controllers/grid/queries/ReadQueryHandler.js

```diff
@@ -0,0 +1,150 @@
+
+	/** @type {Object} */
+	$.pkp.controllers.grid.queries =
+			$.pkp.controllers.grid.queries || { };
+
+
+
+	/**
+	 * @constructor
+	 *
+	 * @extends $.pkp.classes.Handler
+	 *
+	 * @param {jQueryObject} $containerElement The HTML element encapsulating
+	 *  the carousel container.
+	 * @param {Object} options Handler options.
+	 */
+	$.pkp.controllers.grid.queries.ReadQueryHandler =
+			function($containerElement, options) {
+
+		this.fetchNoteFormUrl_ = options.fetchNoteFormUrl;
+		this.fetchParticipantsListUrl_ = options.fetchParticipantsListUrl;
+
+		$containerElement.find('.sprite.add').click(
+				this.callbackWrapper(this.showNoteFormHandler));
+
+		$containerElement.bind('dataChanged', this.callbackWrapper(this.tester_));
+
+		this.loadParticipantsList();
+	};
+	$.pkp.classes.Helper.inherits(
+			$.pkp.controllers.grid.queries.ReadQueryHandler,
+			$.pkp.classes.Handler);
+
+
+	//
+	// Private properties
+	//
+	/**
+	 * The URL to be called to fetch a spotlight item via autocomplete.
+	 * @private
+	 * @type {string?}
+	 */
+	$.pkp.controllers.grid.queries.ReadQueryHandler.
+			prototype.fetchNoteFormUrl_ = null;
+
+
+	/**
+	 * The URL to be called to fetch a list of participants.
+	 * @private
+	 * @type {string?}
+	 */
+	$.pkp.controllers.grid.queries.ReadQueryHandler.
+			prototype.fetchParticipantsListUrl_ = null;
+
+
+	//
+	// Public methods
+	//
+	/**
+	 * Event handler that is called when the suggest username button is clicked.
+	 * @param {HTMLElement} element The checkbox input element.
+	 */
+	$.pkp.controllers.grid.queries.ReadQueryHandler.prototype.
+			showNoteFormHandler = function(element) {
+		$(element).hide();
+		$.get(this.fetchNoteFormUrl_,
+				this.callbackWrapper(this.showFetchedNoteForm_), 'json');
+	};
+
+
+	/**
+	 * Load the participants list.
+	 */
+	$.pkp.controllers.grid.queries.ReadQueryHandler.prototype.
+			loadParticipantsList = function() {
+		$.get(this.fetchParticipantsListUrl_,
+				this.callbackWrapper(this.showFetchedParticipantsList_), 'json');
+	};
+
+
+	//
+	// Private methods
+	//
+	/**
+	 * Event handler that is called when the suggest username button is clicked.
+	 * @param {Object} ajaxContext The AJAX request context.
+	 * @param {Object} jsonData A parsed JSON response object.
+	 * @private
+	 */
+	$.pkp.controllers.grid.queries.ReadQueryHandler.prototype.
+			showFetchedNoteForm_ = function(ajaxContext, jsonData) {
+
+		var processedJsonData = this.handleJson(jsonData),
+				$noteFormContainer = $('#newNotePlaceholder');
+
+		$noteFormContainer.children().remove();
+		$noteFormContainer.append(processedJsonData.content);
+	};
+
+
+	/**
+	 * Event handler that is called when the participants list fetch is complete.
+	 * @param {Object} ajaxContext The AJAX request context.
+	 * @param {Object} jsonData A parsed JSON response object.
+	 * @private
+	 */
+	$.pkp.controllers.grid.queries.ReadQueryHandler.prototype.
+			showFetchedParticipantsList_ = function(ajaxContext, jsonData) {
```

Comment:
In the "read query" modal, there's an "edit" link to change the participants link. If that results in a change to the list, then the list will need to be reloaded here.